internal/resourcestore/resourcestore.go

```diff
@@ -121,7 +132,7 @@ func (rc *ResourceStore) Put(name string, resource IdentifiableCreatable, cleanu
 		rc.resources[name] = r
 	}
 	// make sure the resource hasn't already been added to the store
-	if r.resource != nil || r.cleanupFuncs != nil {
+	if r.wasPut() {
```

Comment:
Looks like this code would benefit from
```suggestion
	if ok && r.wasPut() {
```

Or something similar (as there's no need to check wasPut if we just created a new empty resource.